server/src/main/java/org/elasticsearch/rest/RestHandler.java

```diff
@@ -89,8 +90,8 @@ default boolean allowsUnsafeBuffers() {
         return Collections.emptyList();
     }
 
-    default boolean compatibilityRequired(){
-        return false;
+    default String compatibleWithVersion(){
```

Comment:
all the usages I can change to use Version, the only places where a byte or a string are used would be XContentBuilder or params.
See AbstractRestChannel

Reply:
I can possibly avoid using params() and add an additional property on `RestRequest` that would keep a Version.
What do you think?

Reply:
Adding to `RestRequest` would be in addition to params right ? Since params is still needed for XContent parsers ?

If so, I worry abit about having this info in two places, especially if the are set at different times. If we can ensure that they are set at the same time, and early enough on the `RestRequest` where it couldn't be read by something and then set. I am good with introducing in two places since I think it reads easier.

Reply:
At the moment there is no usecase that requires `params`. XcontenBuilder has a version set in `AbstractRestChannel` where `RestRequest` is available 
```
        builder.setCompatibleMajorVersion(request.getCompatibleApiVersion().major);
```
`XContentParser` - not currently used for compatible api, but will be for `search_type` https://github.com/pgomulka/elasticsearch/pull/17/files#diff-4ca0a565076688b0f2d8206a8aae8d2d
It is being created in `RestRequest`, where an information about `Version` is stored